### PR TITLE
Reset password

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: src/api/services/http/dotcms-http-response.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Response } from '@angular/http'
+
+export class DotCMSHttpResponse{
+
+    private bodyJsonObject:any;
+
+    public constructor( public resp:Response ){
+        console.log('body', resp._body);
```
Comment (fmontes) [inline src/api/services/http/dotcms-http-response.ts]: Extra console.log remove.

File: src/api/services/http/dotcms-http-response.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        this.bodyJsonObject = JSON.parse(resp._body)
+    }
+
+    get i18nMessagesMap():any{
+        return this.bodyJsonObject.i18nMessagesMap;
+    }
+
+    get entity() : any{
```
Comment (fmontes) [inline src/api/services/http/dotcms-http-response.ts]: Spacing: ```get entity():any {```

File: src/api/services/http/dotcms-http-service.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import {Http, Response, Request, Headers, RequestOptionsArgs} from '@angular/http'
```
Comment (fmontes) [inline src/api/services/http/dotcms-http-service.ts]: I don't the benefit of having this service and core-web-service, I don't think we should create a new one, we should just make a good separation of concerns in core-web-service. 

My 2cents.

**Comment by jgambarios** [comment]
Please remove all the unnecessary **console.log** I think it is ok to have them when handling errors but besaid that it is just local debug code and should not be in the repo.

File: src/view/components/common/login/fogot-password-component/fogot-password-component.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import {DotCMSHttpResponse} from "../../../../../api/services/dotcms-http-response";
+import { Router } from '@ngrx/router';
+
+@Component({
+    directives: [MdButton,  MD_INPUT_DIRECTIVES],
+    encapsulation: ViewEncapsulation.Emulated,
+    moduleId: __moduleName, // REQUIRED to use relative path in styleUrls
+    pipes: [],
+    providers: [],
+    selector: 'dot-fogot-password-component',
```
Comment (fmontes) [inline src/view/components/common/login/fogot-password-component/fogot-password-component.ts]: typo

File: src/view/components/common/login/fogot-password-component/fogot-password-component.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    styleUrls: [],
+    templateUrl: ['fogot-password-component.html'],
```
Comment (fmontes) [inline src/view/components/common/login/fogot-password-component/fogot-password-component.ts]: typo ```foRgot```

File: src/view/components/common/login/fogot-password-component/fogot-password-component.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+})
+
+export class FogotPasswordComponent {
```
Comment (fmontes) [inline src/view/components/common/login/fogot-password-component/fogot-password-component.ts]: Typo :)

File: src/view/components/common/login/reset-password-component/reset-password-component.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
+<!--RESET PASSWORD FORM-->
+<div class="login__container forgot-password__container">
+    <div class="col-sm-12">
+        <h2 class="login__title">{{resetPasswordLabel}}</h2>
+        <div class="error-message" [innerHTML]="message"></div>
+            <form>
+                <!--<md-input name="password"
```
Comment (fmontes) [inline src/view/components/common/login/reset-password-component/reset-password-component.html]: Remove commented code.

File: src/api/services/dotcms-events-service.ts [hunk 8]
```diff
@@ -29,8 +31,9 @@ export class DotcmsEventsService {
      * Opens the Websocket connection with the System Events end-point.
      */
     connectWithSocket(): void {
-        this.ws = new $WebSocket(this.protocol + '://' + this.baseUrl + this.endPoint);
-        this.ws.send();
+        let loginUser:User = this.loginService.getLoginUser();
+        this.ws = new $WebSocket(`${this.protocol}://${this.baseUrl}${this.endPoint}?userId=${loginUser.userId}`);
+        this.ws.connect();
```
Comment (fmontes) [inline src/api/services/dotcms-events-service.ts]: Thanks!

**Pull request merged**